packages/frontend/src/pages/new_league.tsx

```diff
@@ -0,0 +1,124 @@
+import { useContext, useState } from "react";
+import { faBan, faFloppyDisk } from "@fortawesome/free-solid-svg-icons";
+import type { LeagueCreation } from "@esp-group-one/types";
+import { Sport } from "@esp-group-one/types/src/utils";
+import { Header } from "../components/page/header";
+import { Form } from "../components/form/index";
+import { Input } from "../components/form/input";
+import { RadioButton } from "../components/form/radio_button";
+import { SelectSport } from "../components/form/select_sports";
+import { Button } from "../components/button";
+import { Icon } from "../components/icon";
+import { API } from "../state/auth";
+
+export function NewLeaguePage() {
+  const [name, setName] = useState("");
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [visibility, setVisibility] = useState<boolean>(true);
+  const api = useContext(API);
+
+  const handleCreateLeague = async () => {
+    try {
+      // Make sure all required fields are filled
+      if (!name || !startDate || !endDate) {
+        throw new Error("Please fill in all required fields.");
+      }
+
+      // Construct the league object
+      const formData: LeagueCreation = {
+        name,
+        // startDate,
+        // endDate,
+        // ownerIds: [ok.user._id],
+        sport: Sport.Tennis,
+        private: !visibility,
+      };
+
+      // Call the API to create a new league
+      await api.league().create(formData);
+
+      // Redirect or perform any other necessary actions upon successful creation
+    } catch (error) {
+      console.error("Error creating league:", error);
+      // Handle error (e.g., display error message)
+    }
+  };
+
+  const handleCancel = () => {
+    // Handle cancellation (e.g., navigate back)
```

Comment:
Look at using `viewNav`:

E.g. put this at the top of the file
```ts
const viewNav = useViewNav()
```

Then you can run

```
viewNav("/leagues")
```